src/main/java/Duke.java

```diff
@@ -1,10 +1,358 @@
+     * It also writes the list of tasks to the local storage.
+     *
+     * @param userCommand     Indicates the instruction (done, delete, etc) to the bot.
+     * @param due             Provides the due date/ time of the task.
+     * @param taskDescription Reflects the description of a task.
+     * @param storage         The local storage for the Tasks as a Storage object.
+     * @param ui              The User Interface object that performs all interactions with the user.
+     * @param tasks           The list of Task that are temporary stored with the bot.
+     * @return Gives a feedback to the user on the operation has performed after a command is given.
+     * @throws EmptyToDoDescriptionException Indicates an empty description that should not be left empty for Todo.
+     * @throws EmptyDescriptionException     Indicates an empty description that should not be left empty.
+     * @throws UnknownCommandException       Indicates the inability of the bot to read the command that is given by the user.
+     */
+    public static String createTask(String userCommand, String due, String taskDescription, Storage storage, Ui ui, TaskList tasks) throws EmptyToDoDescriptionException, EmptyDescriptionException, UnknownCommandException {
+
+        Task t;
+        String typeOfTask = "";
+        if (userCommand.equals("todo")) {
+            if (taskDescription.equals("dummy")) {
+                throw new EmptyToDoDescriptionException("The description of a todo cannot be empty.");
+            }
+            t = new Todo(taskDescription);
+            typeOfTask = "T";
+            tasks.add(t);
+            storage.writeToFile(typeOfTask, "0", taskDescription, t);
+        } else if (userCommand.equals("deadline")) {
+            Date dateDue = convertStringToDate(due);
+            if (taskDescription.equals("dummy")) {
+                throw new EmptyDescriptionException("The description of a deadline cannot be empty.");
+            }
+            t = new Deadline(taskDescription, dateDue);
+            tasks.add(t);
+            typeOfTask = "D";
+            storage.writeToFile(typeOfTask, "0", taskDescription, t);
+        } else if (userCommand.equals("event")) {
+            String[] eventStartEnd = due.split("-", 2);
+            Date start = convertStringToDate(eventStartEnd[0]);
+            Date end = convertStringToDate(eventStartEnd[1]);
+
+            if (taskDescription.equals("dummy")) {
+                throw new EmptyDescriptionException("The description of a event cannot be empty.");
+            }
+            t = new Event(taskDescription, start, end);
+            typeOfTask = "E";
+            tasks.add(t);
+            storage.writeToFile(typeOfTask, "0", taskDescription, t);
+        } else {
+            throw new UnknownCommandException("I'm sorry, but I don't know what that means :-(");
+        }
+
+        int taskCount = Task.getTaskCount();
+        return ui.newTaskAdded(t, taskCount);
     }
 }
```

Comment:
Other than that, your Duke implementation looks clean, neat and good! 

I can learn some stuff from you, such as breaking down the Error classes into more OOP. All the best. Cheers!